app/src/main/java/com/better/alarm/presenter/AlarmsListActivity.kt

```diff
@@ -202,6 +208,28 @@ class AlarmsListActivity : AppCompatActivity() {
             }.apply { }
     }
 
+    private fun showMuteAlert() {
+        val audio: AudioManager = this.getSystemService(AUDIO_SERVICE) as AudioManager
+        if(audio.getStreamVolume(AudioManager.STREAM_ALARM)<=1) {
+            //implementation of snackbar
+//            Toast.makeText(this,"" + audio.getStreamVolume(AudioManager.STREAM_ALARM) , Toast.LENGTH_LONG).show();
+            val rootView = findViewById<View>(android.R.id.content)
+            val toastText = "Warning: Alarm volume is low"
+            val volumeSnack = Snackbar.make(rootView, toastText, Snackbar.LENGTH_INDEFINITE)
+
+
+            volumeSnack.setAction("Test Alarm") {
+                val notification: Uri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_ALARM)
+                val r = RingtoneManager.getRingtone(applicationContext, notification)
+                r.play()
+                Executors.newSingleThreadScheduledExecutor().schedule({
```

Comment:
I would prefer if this would not be automatically turned off after 2 seconds. How about changing the "Test" button into the "Stop" button? This will allow the user to use volume buttons to change the volume.

+ and - buttons can also be displayed and used to change the volume while testing.

And, of course, stop the playback in onPause or using Snackbar callbacks.